lib/services/runtime/handlers/message/lib/selectVariant.ts

```diff
@@ -0,0 +1,35 @@
+import { CompiledResponseMessage } from '@voiceflow/dtos';
+
+import { BaseCondition } from './conditions/base.condition';
+
+interface VariantConditionPair {
+  variant: CompiledResponseMessage;
+  condition: BaseCondition | null;
+}
+
+export async function selectVariant(variants: VariantConditionPair[]): Promise<CompiledResponseMessage> {
+  const conditionedVariants: VariantConditionPair[] = [];
+  const unconditionedVariants: VariantConditionPair[] = [];
+
+  variants.forEach((pair) => {
+    if (pair.condition) {
+      conditionedVariants.push(pair);
+    } else {
+      unconditionedVariants.push(pair);
+    }
+  });
```

Comment:
```suggestion
  const [conditionedVariants, unconditionedVariants]: VariantConditionPair[][] = variants.reduce(
    ([conditioned, unconditioned], pair) => {
      if (pair.condition) {
        conditionedVariants.push(pair);
      } else {
        unconditionedVariants.push(pair);
      }
      return [conditioned, unconditioned];
    },
    [[], []]
  );
```

immutable with reduce. i find it an odd anti-pattern to use `const` with an array that gets mutated. so the other option is to keep what you have and use `let` over `const` for understanding at a glance that the arrays are mutable.